### PR TITLE
feat: update paypal content

### PR DESCRIPTION
**Pull request by radireddy**
change paypal content for US region

ref: MANAGER-7846

Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-7846
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/billing/src/payment/method/add/index.html [hunk 1]
```diff
@@ -85,10 +85,16 @@
...
+                                <p data-ng-switch-when="PAYPAL">
+                                    <span
+                                        data-ng-if="$ctrl.user.ovhSubsidiary === 'US'"
+                                        data-translate="billing_payment_method_add_selected_paypal_redirect_US"
+                                    ></span>
+                                    <span
+                                        data-ng-if="$ctrl.user.ovhSubsidiary !== 'US'"
+                                        data-translate="billing_payment_method_add_selected_paypal_redirect"
+                                    ></span>
+                                </p>
```
Comment (anooparveti) [inline packages/manager/modules/billing/src/payment/method/add/index.html:97]: `<p 
    data-ng-switch-when="PAYPAL" 
    data-ng-bind=":: ($ctrl.user.ovhSubsidiary === 'US' ? 'billing_payment_method_add_selected_paypal_redirect_US' : 'billing_payment_method_add_selected_paypal_redirect') | translate" >
</p>`

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/directive.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default {
     integrationCtrl.loading.init = true;
 
     // insert paypal checkout.js script if not yet loaded
-    if (!document.getElementById(PAYPAL_SCRIPT.id)) {
+    if (typeof paypal === 'undefined') {
```
Comment (frenautvh) [inline packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/directive.js:42]: any reason for not using
```js
if (!paypal)
````
or
```js
if (paypal === undefined)
```

**Review by frenautvh — DISMISSED** [review]

File: packages/manager/modules/billing/src/payment/method/add/index.html (hunk 1 above)
Comment (anooparveti) [inline packages/manager/modules/billing/src/payment/method/add/index.html:97]: `<p 
         data-ng-switch-when="PAYPAL" 
         data-ng-bind=":: ($ctrl.user.ovhSubsidiary === 'US' ? 'billing_payment_method_add_selected_paypal_redirect_US' : 'billing_payment_method_add_selected_paypal_redirect') | translate"></p>`

Comment (radireddy) [reply packages/manager/modules/billing/src/payment/method/add/index.html:97]: Yes this works. Just to keep it simple and easy to understand, I have split it. At the end both offer same benefits.

**Review by anooparveti — DISMISSED** [review]

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/directive.js (hunk 2 above)
Comment (radireddy) [reply packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/directive.js:42]: No specific reason. All these 3 approaches works.

**Review by anooparveti — APPROVED** [review]

File: packages/components/ng-ovh-payment-method/src/components/register/index.html [hunk 3]
```diff
@@ -70,4 +72,40 @@
...
         </oui-message>
     </div>
+    <div
+        data-ng-if="$ctrl.showSetAsDefaultChoice && $ctrl.user.ovhSubsidiary === 'US'"
+    >
+        <p
+            data-ng-if="$ctrl.registeredPaymentMethods.length === 0 && $ctrl.automaticDefault"
+        >
+            <span
+                data-ng-if="$ctrl.model.selectedPaymentMethodType.type.paymentType === 'PAYPAL'"
```
Comment (varun257) [inline packages/components/ng-ovh-payment-method/src/components/register/index.html:82]: `PAYPAL` can be a constant here

File: packages/components/ng-ovh-payment-method/src/components/register/index.html [hunk 4]
```diff
@@ -41,7 +41,9 @@
     <div data-ng-transclude="explanationText"></div>
 
     <!-- set as default -->
-    <div data-ng-if="$ctrl.showSetAsDefaultChoice">
+    <div
+        data-ng-if="$ctrl.showSetAsDefaultChoice && $ctrl.user.ovhSubsidiary !== 'US'"
```
Comment (mohammed-zahaf) [inline packages/components/ng-ovh-payment-method/src/components/register/index.html:45]: May be we can use:
```suggestion
        data-ng-if="$ctrl.showSetAsDefaultChoice && $ctrl.coreConfig.isRegion('US')"
```

File: packages/components/ng-ovh-payment-method/src/components/register/index.html [hunk 5]
```diff
@@ -70,4 +72,40 @@
             ></span>
         </oui-message>
     </div>
+    <div
+        data-ng-if="$ctrl.showSetAsDefaultChoice && $ctrl.user.ovhSubsidiary === 'US'"
```
Comment (mohammed-zahaf) [inline packages/components/ng-ovh-payment-method/src/components/register/index.html:76]:
```suggestion
        data-ng-if="$ctrl.showSetAsDefaultChoice && $ctrl.coreConfig.isRegion('US')"
```

File: packages/manager/modules/billing/src/payment/method/add/index.html [hunk 6]
```diff
@@ -85,10 +85,16 @@
                                     >
                                     </span>
                                 </p>
-                                <p
-                                    data-ng-switch-when="PAYPAL"
-                                    data-translate="billing_payment_method_add_selected_paypal_redirect"
-                                ></p>
+                                <p data-ng-switch-when="PAYPAL">
+                                    <span
+                                        data-ng-if="$ctrl.user.ovhSubsidiary === 'US'"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/billing/src/payment/method/add/index.html:90]:
```suggestion
                                        data-ng-if="$ctrl.coreConfig.isRegion('US')"
```

File: packages/manager/modules/billing/src/payment/method/add/index.html [hunk 7]
```diff
@@ -85,10 +85,16 @@
...
+                                        data-translate="billing_payment_method_add_selected_paypal_redirect_US"
+                                    ></span>
+                                    <span
+                                        data-ng-if="$ctrl.user.ovhSubsidiary !== 'US'"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/billing/src/payment/method/add/index.html:94]:
```suggestion
                                        data-ng-if="$ctrl.coreConfig.isRegion('US')"
```

File: packages/manager/modules/billing/src/payment/method/add/controller.js [hunk 8]
```diff
@@ -13,13 +13,15 @@ export default class BillingPaymentMethodAddCtrl {
...
     OVH_PAYMENT_METHOD_INTEGRATION_TYPE,
+    coreConfig,
   ) {
     this.$translate = $translate;
     this.Alerter = Alerter;
     this.ovhContacts = ovhContacts;
     this.ovhPaymentMethod = ovhPaymentMethod;
     this.OVH_PAYMENT_METHOD_TYPE = OVH_PAYMENT_METHOD_TYPE;
     this.OVH_PAYMENT_METHOD_INTEGRATION_TYPE = OVH_PAYMENT_METHOD_INTEGRATION_TYPE;
+    this.user = coreConfig.getUser();
```
Comment (mohammed-zahaf) [inline packages/manager/modules/billing/src/payment/method/add/controller.js:24]: Regarding my suggestions on HTML templates:

```suggestion
    this.coreConfig= coreConfig;
```

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/directive.js (hunk 2 above)
Comment (radireddy) [reply packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/directive.js]: done

**Review by anooparveti — DISMISSED** [review]

**Review by varun257 — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by JeremyDec — DISMISSED** [review]

File: packages/manager/modules/billing/src/payment/method/add/controller.js [hunk 9]
```diff
@@ -24,6 +25,7 @@ export default class BillingPaymentMethodAddCtrl {
     this.ovhPaymentMethodHelper = ovhPaymentMethodHelper;
     this.OVH_PAYMENT_METHOD_TYPE = OVH_PAYMENT_METHOD_TYPE;
     this.OVH_PAYMENT_METHOD_INTEGRATION_TYPE = OVH_PAYMENT_METHOD_INTEGRATION_TYPE;
+    this.coreConfig = coreConfig;
```
Comment (antleblanc) [inline packages/manager/modules/billing/src/payment/method/add/controller.js:28]: **nitick:** `@ovh-ux/ng-ovh-payment-method` which is a public component shouldn't use `coreConfig` (from `@ovh-ux/manager-core`) which is a private package.

Comment (antleblanc) [reply packages/manager/modules/billing/src/payment/method/add/controller.js:28]: Will be addressed in a separate pull request.

**Review by antleblanc — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Pull request merged**